internal/rest/k8s_utils.go

```diff
@@ -152,7 +174,7 @@ func (l *leader) SyncObjectStatuses() {
 			publisher.UpdateL4LBStatus(allServiceLBUpdateOptions, true)
 		}
 		publisher.UpdateRouteIngressStatus(allIngressUpdateOptions, true)
-		if !lib.GetLayer7Only() {
+		if !lib.GetLayer7Only() && !lib.UseServicesAPI() {
```

Comment:
@DixitAakash : Can we optimize this as "if serivceAPi {} else if !Layer7{}

Reply:
@akshayhavile this function call **publisher.UpdateRouteIngressStatus(allIngressUpdateOptions, true)** in line 176 needs to be moved down for optimization.
```
if lib.UseServicesAPI() {
	publisher.UpdateSvcApiGatewayStatusAddress(allGatewayUpdateOptions, true)
	publisher.UpdateL4LBStatus(allServiceLBUpdateOptions, true)
} else if !lib.GetLayer7Only() {
	publisher.UpdateL4LBStatus(allServiceLBUpdateOptions, true)
}
publisher.UpdateRouteIngressStatus(allIngressUpdateOptions, true)
```

Reply:
Optimization done.